### PR TITLE
feat: Authentication can be optionally disabled

### PR DESCRIPTION
**Pull request by kumare3**
# TL;DR
FlyteConsole enables authentication by default. This has a weird side-effect in Sandbox, where the console enables secure cookies, even if we actually do not have authN. in this situation, Chrome does not allow CORS with a wild card of `*`. This PR allows disabling authN and which should allow working with CORs

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [-] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
Disable credentials in case when authn is not used / required. This will allow CORs wildcard. Particularly helpful in sandbox testing.

## Tracking Issue
https://github.com/lyft/flyte/issues/416

## Follow-up issue
_NA_

### CODE REVIEW

File: env.js [hunk 1]
```diff
@@ -39,6 +41,7 @@ module.exports = {
         BASE_URL,
         CORS_PROXY_PREFIX,
         NODE_ENV,
-        STATUS_URL
+        STATUS_URL,
```
Comment (schottra) [inline env.js:44]: Two small things:
1. These are sorted alphabetically, would prefer if you can keep that.
2. It's not documented, but the way this object works is that items in `processEnv` will be sent to the client, and items outside of it are available on the server. I think this new variable should be in both places, so can you add it at both levels of the object?

File: src/models/AdminEntity/AdminEntity.ts [hunk 2]
```diff
@@ -15,6 +16,16 @@ import {
     logProtoResponse
 } from './utils';
 
+/**
```
[the remark this answers is not in the document]
Comment (kumare3) [reply src/models/AdminEntity/AdminEntity.ts]: done
